tests/scripts/expect/posix-max-power-table.exp

```diff
@@ -31,12 +31,6 @@ source "tests/scripts/expect/_common.exp"
 
 # allows 11-25 and forbidden 26
 spawn_node 1 "rcp" "spinel+hdlc+uart://$env(OT_SIMULATION_APPS)/ncp/ot-rcp?max-power-table=11,12,13,14,15,16,17,18,19,20,21,22,23,24,-1,0x7f&forkpty-arg=1"
-send "channel supported\n"
```

Comment:
I think we should avoid change existing behaviors if not necessary. Even if OPENTHREAD_POSIX_CONFIG_CONFIGURATION_FILE_ENABLE is enabled, it should support starting Thread stack without a configuration file or empty configuration file and behave like the feature is not enabled.

Reply:
I have updated the code to support the behaviors you mentioned.